src/Analyzer.Types/IEvaluation.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Templates.Analyzer.Types
+{
+    /// <summary>
+    /// Describes the evaluation of a TemplateAnalyzer rule against an ARM template
+    /// </summary>
+    public interface IEvaluation
+    {
+        /// <summary>
+        /// Gets the name of the rule this evaluation is for.
+        /// </summary>
+        public string RuleName { get; }
+
+        /// <summary>
+        /// Gets the description of the rule this evaluation is for.
+        /// </summary>
+        public string RuleDescription { get; }
+
+        /// <summary>
+        /// Gets the recommendation for addressing this evaluation.
+        /// </summary>
+        public string Recommendation { get; }
+
+        /// <summary>
+        /// Gets the Uri where help for this evaluation can be found.
+        /// </summary>
+        public string HelpUri { get; }
+
+        /// <summary>
+        /// Gets the identifier of the file this evaluation is for.
+        /// </summary>
+        public string FileIdentifier { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether or not the rule for this evaluation passed.
+        /// </summary>
+        public bool Passed { get; }
+
+        /// <summary>
+        /// Gets the collections of evaluations from this evaluation.
+        /// </summary>
+        public IEnumerable<IEvaluation> Evaluations { get; }
+
+        /// <summary>
+        /// Gets the collections of results from this evaluation.
+        /// </summary>
+        public IEnumerable<IResult> Results { get; }
+
+        /// <summary>
+        /// Gets wether this evaluation has corresponding results.
+        /// </summary>
+        /// <returns>Whether this evaluation has corresponding results</returns>
+        public bool HasResults()
+        {
+            return Results != null && Results.Count() > 0;
```

Comment:
I don't know if it matters right now, but this will have the same IEnumerable iteration issue we discussed earlier - `.Count()` will iterate over it, so the next time it's referenced, it could reevaluate everything again.

Reply:
I tried making it a property instead and cache the bool, but it won't work since the class is an interface.

Reply:
Maybe this should be moved into JsonRuleEvaluation.  We can take the decision later.